### ISSUE TITLE
`jj init --git-repo` fails on bare repositories

### ISSUE THREAD

**Issue by orip** [martinvonz/jj#1249]
## Description

Adding a `--bare` option would be great, or auto-detection of course.

## Steps to Reproduce the Problem

```
% git clone --bare https://github.com/githubtraining/hellogitworld.git                  
Cloning into bare repository 'hellogitworld.git'...
remote: Enumerating objects: 306, done.
remote: Total 306 (delta 0), reused 0 (delta 0), pack-reused 306
Receiving objects: 100% (306/306), 95.81 KiB | 560.00 KiB/s, done.
Resolving deltas: 100% (69/69), done.

% jj init --git-repo=hellogitworld.git foo
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { code: -3, klass: 2, message: "failed to resolve path '/private/tmp/hellogitworld.git/.git': No such file or directory" }', lib/src/git_backend.rs:84:75
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

## Specifications

- Platform: MacOS
- Version: 0.6.1

**Comment by orip** [comment]
My personal scenario is a bare repo with several worktrees. I was planning on trying `jj` as if it were another worktree.

**Comment by xelxebar** [comment]
I believe `GIT_COMMON_DIR` and [commondir][0] need to be respected for this. My use case is similar to the original poster's and I'm seeing this:

```
$ jj init --git-repo=.
Failed to add `.jj/` to /home/x/devel/.repos/5b3b207f-248f-4ed1-a65f-bd8244a97e06.git/worktrees/dyalog-play/info/exclude because it doesn't exist
Initialized repo in "."
```

However,

```
$ cat .git
gitdir: /home/x/devel/.repos/5b3b207f-248f-4ed1-a65f-bd8244a97e06.git/worktrees/dyalog-play
$ awk '{print $2}' .git | xargs ls
COMMIT_EDITMSG  FETCH_HEAD  HEAD  ORIG_HEAD  commondir  gitdir  index  logs
$ awk '{print $2}' .git | xargs -I{} cat '{}/commondir'
../..
$ awk '{print $2}' .git | xargs -I{} ls '{}/../../info/exclude'
/home/x/devel/.repos/5b3b207f-248f-4ed1-a65f-bd8244a97e06.git/worktrees/dyalog-play/../../info/exclude
```

So it looks like the `gitdir` ref is correctly resolving, but we just need to follow the `commondir` indirection. 

Also, FWIW, the particular workaround isn't quite working in my case. After editing `git_target`,

```
$ jj st
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', lib/src/git_backend.rs:90:79
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

$ xargs ls <.jj/repo/store/git_target
COMMIT_EDITMSG  FETCH_HEAD  HEAD  ORIG_HEAD  commondir  gitdir  index  logs
```


[0]:https://git-scm.com/docs/gitrepository-layout#Documentation/gitrepository-layout.txt-commondir

**Comment by yuja** [comment]
Should be addressed by 10cdef49.
